client/object_search.go

```diff
@@ -175,6 +175,27 @@ func (x *ObjectListReader) Read(buf []oid.ID) (int, bool) {
 	}
 }
 
+// Iterate iterates over the list of found object identifiers.
+// Returns an error if object can't be read.
+//
+// Panics if buf has zero length.
```

Comment:
Non actual here, guess you meant `handler is nil`?

Also lets mention that iterator breaks on f's true return?

Maybe we make it a function: it uses only public methods of the type.